graylog2-server/src/main/java/org/graylog2/rest/resources/system/inputs/InputsResource.java

```diff
@@ -0,0 +1,208 @@
+            final String newId = inputService.save(input);
+            final URI inputUri = getUriBuilderToSelf().path(InputsResource.class)
+                    .path("{inputId}")
+                    .build(newId);
+
+            return Response.created(inputUri).entity(InputCreated.create(newId)).build();
+        } catch (NoSuchInputTypeException e) {
+            LOG.error("There is no such input type registered.", e);
+            throw new NotFoundException("There is no such input type registered.", e);
+        } catch (ConfigurationException e) {
+            LOG.error("Missing or invalid input configuration.", e);
+            throw new BadRequestException("Missing or invalid input configuration.", e);
+        }
+
+    }
+
+    @DELETE
+    @Timed
+    @Path("/{inputId}")
+    @ApiOperation(value = "Terminate input on this node")
+    @ApiResponses(value = {
+            @ApiResponse(code = 404, message = "No such input on this node.")
+    })
+    public void terminate(@ApiParam(name = "inputId", required = true) @PathParam("inputId") String inputId) throws org.graylog2.database.NotFoundException {
+        final Input input = inputService.find(inputId);
+        inputService.destroy(input);
+    }
+
+    @PUT
+    @Timed
+    @Path("/{inputId}")
+    @ApiOperation(
+            value = "Update input on this node",
+            response = InputCreated.class
+    )
+    @ApiResponses(value = {
+            @ApiResponse(code = 404, message = "No such input on this node."),
+            @ApiResponse(code = 400, message = "Missing or invalid input configuration.")
+    })
+    public Response update(@ApiParam(name = "JSON body", required = true) @Valid @NotNull InputCreateRequest lr,
+                           @ApiParam(name = "inputId", required = true) @PathParam("inputId") String inputId) throws org.graylog2.database.NotFoundException, NoSuchInputTypeException, ConfigurationException, ValidationException {
+        checkPermission(RestPermissions.INPUTS_EDIT, inputId);
+
+        final Input input = inputService.find(inputId);
+
+        final Map<String, Object> mergedInput = inputService.getMessageInput(input).asMap();
+        final MessageInput messageInput = messageInputFactory.create(lr, getCurrentUser().getName(), lr.node());
+
+        messageInput.checkConfiguration();
+
+        mergedInput.putAll(messageInput.asMap());
+
+        final Input newInput = inputService.create(input.getId(), mergedInput);
+        inputService.save(newInput);
+
+        final URI inputUri = getUriBuilderToSelf().path(InputsResource.class)
+                .path("{inputId}")
+                .build(input.getId());
+
+        return Response.created(inputUri).entity(InputCreated.create(input.getId())).build();
+    }
+
+    private InputSummary getInputSummary(Input input) {
+        return InputSummary.create(input.getTitle(),
+                input.isGlobal(),
+                input.getType(),
```

Comment:
Totally right. Gonna change the value of these back to the one we had. Good find, thanks! :+1: